src/general_tally.cpp

```diff
@@ -0,0 +1,210 @@
+    return;
+  }
+
+  const double Et = mat.Et(p.E());
+  const double collision_score = particle_base_score(p, mat) / Et;
+
+#ifdef ABEILLE_USE_OMP
+#pragma omp atomic
+#endif
+  tally_gen_score(indexes) += collision_score;
+}
+
+void GeneralTally::score_flight(const Particle& p, const Tracker& trkr,
+                                double d_flight, MaterialHelper& mat) {
+  if (!(estimator_ == GeneralTally::Estimator::TrackLength)) {
+    return;
+  }
+
+  // get the energy index
+  std::size_t index_E;
+  if (energy_in_) {
+    // get the energy_index if we in the bounds
+    std::optional<std::size_t> E_indx = energy_in_->get_index(p.E());
+    if (E_indx.has_value()) {
+      index_E = E_indx.value();
+    } else {
+      return;
+    }
+  }
+
+  double flight_score = particle_base_score(p, mat);
+
+  // get the all the bins' index along with the distance travelled by the
+  // particle
+  std::vector<TracklengthDistance> pos_indexes_ =
+      position_filter_->get_indices_tracklength(trkr, d_flight);
+
+  for (size_t iter = 0; iter < pos_indexes_.size(); iter++) {
+    StaticVector4 all_indexes_(pos_indexes_[iter].indexes_.begin(),
+                               pos_indexes_[iter].indexes_.end());
+    const double d_ = pos_indexes_[iter].distance_in_bin;
+
+    if (energy_in_) {
+      all_indexes_.insert(all_indexes_.begin(), index_E);
+    }
+
+#ifdef ABEILLE_USE_OMP
+#pragma omp atomic
+#endif
+    tally_gen_score(all_indexes_) += d_ * flight_score;
+  }
+}
+
+std::shared_ptr<GeneralTally> make_general_tally(const YAML::Node& node) {
+  // Check the name of the tally is given or not.
+  if (!node["name"]) {
+    fatal_error("No valid name is provided.");
+  }
+  std::string general_tally_name = node["name"].as<std::string>();
+
+  // Check the estimator is given or not.
+  if (!node["estimator"]) {
+    fatal_error(
+        "No, estimator is given for \"" + general_tally_name +
+        "\", therefore, collision estimator will be taken, by default.");
+  }
+  std::string estimator_name = node["estimator"].as<std::string>();
+
+  // Check for the quantity
+  std::string given_quantity = "";
+  if (!node["quantity"]) {
+    fatal_error("No quantity is given for " + general_tally_name + " tally.");
+  }
+  given_quantity = node["quantity"].as<std::string>();
+  GeneralTally::Quantity quant;
+
+  bool found_quantity = false;
+  if (given_quantity == "flux") {
+    quant = GeneralTally::Quantity::Flux;
+    found_quantity = true;
+  }
+
+  if (given_quantity == "fission") {
+    quant = GeneralTally::Quantity::Fission;
+    found_quantity = true;
+  }
+  if (given_quantity == "absorption") {
+    quant = GeneralTally::Quantity::Absorption;
+    found_quantity = true;
+  }
+  if (given_quantity == "elastic") {
+    quant = GeneralTally::Quantity::Elastic;
+    found_quantity = true;
+  }
+
+  if (found_quantity == false) {
+    fatal_error("For " + general_tally_name +
+                " tally, a unknown quantity is given.");
+  }
```

Comment:
Should be chained else if statements with a final else for fatal error when the quantity doesn't match a known value.